Sources/Commands/SwiftPackageTool.swift

```diff
@@ -183,7 +183,24 @@ public struct SwiftPackageTool: SwiftTool {
                     return try get(manifest, manifestParser: parseManifest)
                 }
             }
-        
+
+            func findDirectoryReferences(path: String) throws -> [String] {
+                let rootDirectories = walk(path, recursively: false)
+                let rootDirectoriesToConsider = rootDirectories.filter {
+                    if $0.hasSuffix(".xcodeproj") { return false }
+                    if $0.hasSuffix(".playground") { return false }
+                    if $0.basename.hasPrefix(".") { return false }
+                    return $0.isDirectory
+                }
+
+                return rootDirectoriesToConsider.filter {
+                    let directoriesWithSwift = walk($0).filter {
+                        $0.hasSuffix(".swift")
```

Comment:
I think if this code is going to be used it should be in Xcodeproj module.
Also along with swift we also support c languages, look at the `SupportedLanguageExtension` enum in `PackageModel/Sources.swift`